### PR TITLE
DistanceFromSame 

### PR DESCRIPTION
**Pull request by Ilanox**
Added DistanceFromSame option in the StructureLocation section, After the Discord Server wanted.

Tried to use the new BottonFill but reversed it, Don't mind it

### CODE REVIEW

File: src/main/java/com/ryandw11/structure/commands/cstruct/TestSpawnCommand.java [hunk 1]
```diff
@@ -198,6 +198,9 @@ void canSpawn(Player p, Structure structure, Block block, Chunk chunk) {
         if (!CustomStructures.getInstance().getStructureHandler().validDistance(structure, block.getLocation()))
             quickSendMessage(p, "&cFailed Distance Limitation test! Cannot spawn this close to another structure!");
 
+        if (!CustomStructures.getInstance().getStructureHandler().validSameDistance(structure, block.getLocation()))
+            quickSendMessage(p, "&cFailed Distance Limitation test! Cannot spawn this close to the same structure!");
+
```
Comment (tatanpoker09) [inline src/main/java/com/ryandw11/structure/commands/cstruct/TestSpawnCommand.java:203]: I'm not that familiar with the codebase, but I'm wondering why this is checked both here and structure.canSpawn(). Couldn't we just call structure.canSpawn() to check all conditions? I understand that would generate problems to give separate error messages, but going forward might make it cleaner.
Then again, sorry if this comment makes no sense as I am really unfamiliar with the codebase

File: src/main/java/com/ryandw11/structure/structure/StructureHandler.java [hunk 2]
```diff
@@ -166,6 +168,25 @@ public boolean validDistance(Structure struct, Location location) {
...
+            for (Map.Entry<Pair<Location, Long>, Structure> entry : spawnedStructures.entrySet()) {
+
+                if (entry.getKey().getLeft().getWorld() != location.getWorld()) continue;
+
+                if (entry.getKey().getLeft().distance(location) < closest)
+
+                    if (Objects.equals(entry.getValue().getName(), struct.getName())) {
+                        closest = entry.getKey().getLeft().distance(location);
+                    }
+            }
```
Comment (tatanpoker09) [inline src/main/java/com/ryandw11/structure/structure/StructureHandler.java]: Maybe keep the return early pattern if we're already using it before to be consistent.
```suggestion
            for (Map.Entry<Pair<Location, Long>, Structure> entry : spawnedStructures.entrySet()) {

                if (entry.getKey().getLeft().getWorld() != location.getWorld()) continue;
                
                if (!Objects.equals(entry.getValue().getName(), struct.getName())) continue;

                if (entry.getKey().getLeft().distance(location) < closest){
                        closest = entry.getKey().getLeft().distance(location);
                }
            }
```

File: src/main/java/com/ryandw11/structure/commands/cstruct/TestSpawnCommand.java (hunk 1 above)
Comment (Ilanox) [reply src/main/java/com/ryandw11/structure/commands/cstruct/TestSpawnCommand.java:203]: I just added the DistanceFromSame option, I don't really want to change any of the code of ryandw11..

**Review by ryandw11 — APPROVED** [review]
Looks good. Although for any future PRs, don't base them off of any other PRs unless it directly depends on it. Thanks for your contributions.

**Pull request merged**